controllers/configuration_controller.go

```diff
@@ -1279,3 +1331,32 @@ func (meta *TFConfigurationMeta) RenderConfiguration(configuration *v1beta2.Conf
 		return "", nil, errors.New("Unsupported Configuration Type")
 	}
 }
+
+// GetGitCredentialsSecret will get the secret containing the SSH private key & known_hosts
+func GetGitCredentialsSecret(ctx context.Context, k8sClient client.Client, secretRef *v1.SecretReference) (*v1.Secret, error) {
+	secret := &v1.Secret{}
+
+	namespacedName := client.ObjectKey{Name: secretRef.Name, Namespace: secretRef.Namespace}
+	err := k8sClient.Get(ctx, namespacedName, secret)
+	if err != nil {
+		errMsg := "Failed to get git credentials secret"
+		klog.ErrorS(err, errMsg, "Name", secretRef.Name, "Namespace", secretRef.Namespace)
+		return nil, errors.Wrap(err, errMsg)
+	}
+
+	keyNotFoundErrMsg := "key not in secret"
+	keyNotFoundErr := errors.New(keyNotFoundErrMsg)
+
+	if _, ok := secret.Data[v1.SSHAuthPrivateKey]; !ok {
+		errMsg := fmt.Sprintf("'%s' not in git credentials secret", v1.SSHAuthPrivateKey)
+		klog.ErrorS(keyNotFoundErr, errMsg, "Name", secretRef.Name, "Namespace", secretRef.Namespace)
+		return nil, errors.Wrap(keyNotFoundErr, errMsg)
+	}
+	if _, ok := secret.Data["known_hosts"]; !ok {
+		errMsg := "'known_hosts' not in git credentials secret"
+		klog.ErrorS(keyNotFoundErr, errMsg, "Name", secretRef.Name, "Namespace", secretRef.Namespace)
+		return nil, errors.Wrap(keyNotFoundErr, errMsg)
+	}
```

Comment:
1. Please make "known_hosts" a const string, you can both use it in controller and test.
2. Make the detect a loop like
```go
needSecretKeys:=[]string{v1.SSHAuthPrivateKey, GitCredsKnownHosts}
for _,key:=range needSecretKeys{
}
```
3. reduce the message wrap, just return `errors.Errorf("%s key not found in git credentials secret", key) `is ok

Reply:
done